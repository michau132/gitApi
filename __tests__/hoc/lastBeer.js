import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LastBeer from '../../src/hoc/lastBeer';

const component = () => <h1>Hi there</h1>;
const ConditionalHOC = LastBeer(component);
const props = { lastBeer: true };

describe('testing LastBeer HOC', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ConditionalHOC {...props} />);
  });
  test('should render component when loading is set to true', () => {
    expect(wrapper).toBeDefined();
  });

  test('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
