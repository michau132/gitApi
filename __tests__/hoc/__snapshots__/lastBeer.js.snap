// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing LastBeer HOC should match snapshot 1`] = `
<Fragment>
  <component
    lastBeer={true}
  />
  <div>
    <h2>
      Sorry there is no more beer
    </h2>
    <styled.div>
      <Image
        alt="no beer"
        src="https://previews.123rf.com/images/alexutemov/alexutemov1605/alexutemov160500656/56958969-stop-drink-no-beer-design-and-vector-illustration-stop-drink-stop-drink-alcohol-glass-no-sign-beer-s.jpg"
      />
    </styled.div>
  </div>
</Fragment>
`;
